src/pages/Chat/Chat/index.tsx

```diff
@@ -0,0 +1,326 @@
+
+    if (newestDate) {
+      chatAPI.loadNewMessages(chatRef, newestDate).then((loadedMessages) => {
+        const filteredLoadedMessages = loadedMessages.docs.filter((doc) => doc.data().sender.id !== currentUserRef.id);
+        addMessages(filteredLoadedMessages);
+      });
+    }
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [shouldUpdate]);
+
+  const loadMessages = useCallback((lastMessage?: IMessage): void => {
+    if (!chatRef) return;
+
+    const lastDate = lastMessage ? lastMessage.createdAt as Date : undefined;
+
+    chatAPI.loadMessages(chatRef, 10, lastDate)
+      .then((loadedMessages) => {
+        addMessages(loadedMessages.docs);
+        setIsLoadingEarlier(false);
+        firstLoad.current = false;
+      })
+      .catch(() => {
+        Alert.alert(
+          'Erro!',
+          'Ocorreu um erro no carregamento das informações.\n'
+              + 'Retornando para a página anterior.',
+          [
+            {
+              text: 'Ok',
+              onPress: () => navigation.goBack(),
+            },
+          ],
+          {
+            cancelable: false,
+          },
+        );
+      });
+  }, [chatRef, navigation, addMessages]);
+
+  const updateTyping = (): void => {
+    // onInputTextChanged triggers when first load
+    if (ignoreFirstOnInputTextChanged.current === true) {
+      ignoreFirstOnInputTextChanged.current = false;
+      return;
+    }
+
+    // Updates db if changed selfIsTypingState
+    if (selfIsTyping.current === false) {
+      if (chatRef) chatAPI.setIsTyping(chatRef, currentUserRef, true);
+    }
+
+    selfIsTyping.current = true;
+
+    if (typingTimeoutFunction.current) {
+      clearTimeout(typingTimeoutFunction.current);
+    }
+
+    typingTimeoutFunction.current = setTimeout(() => {
+      selfIsTyping.current = false;
+      if (chatRef) chatAPI.setIsTyping(chatRef, currentUserRef, false);
+    }, 1000);
+  };
+
+  // Effect to load chatRef from routeChatUID or targetUserUID
+  useEffect(() => {
+    if (routeChatUID !== undefined) {
+      setChatRef(chatAPI.chatDocument(routeChatUID));
+      chatAPI.markChatMessagesAsSeemByUser(routeChatUID, currentUserRef);
+    } else if (routeChatUID === undefined && targetUserUID !== undefined) {
+      chatAPI.getChatByTarget(currentUserRef, userAPI.userDocument(targetUserUID)).then((chat) => {
+        if (chat?.ref) {
+          setChatRef(chat.ref);
+          chatAPI.markChatMessagesAsSeemByUser(chat.ref.id, currentUserRef);
+        } else {
+          setMessages([]);
+        }
+      });
+    }
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  // Initial load messages when chatRef is set
+  useEffect(() => loadMessages(), [loadMessages, chatRef]);
+
+  // Get real time chat updates
+  useEffect(() => {
+    if (!chatRef) return () => null;
+    const unsubscribe = chatRef.collection('updatedAt')
+      .onSnapshot(() => {
+        setShouldUpdate((previous) => previous + 1);
```

Comment:
This one is a little more complex, so maybe an issue for future tests would be fitting. Why use a number setState to trigger an useEffect? Couldn't the useEffect be converted to a callback and called in this line and on the initial load useEffect?